oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/reactive/function/client/OAuth2AuthorizedClientResolver.java

```diff
@@ -133,7 +133,7 @@ public void setClientCredentialsTokenResponseClient(
 			});
 }
 
-	private Mono<? extends OAuth2AuthorizedClient> clientCredentials(
+	public Mono<OAuth2AuthorizedClient> clientCredentials(
```

Comment:
I don't think we should expose a public API here. What you can do instead of:

`when(this.oAuth2AuthorizedClientResolver.clientCredentials(any(), any(), any())).thenReturn(Mono.just(newAuthorizedClient));`

try this

Mock `ReactiveOAuth2AccessTokenResponseClient<OAuth2ClientCredentialsGrantRequest>` and set it via  `OAuth2AuthorizedClientResolver.setClientCredentialsTokenResponseClient()` and you'll also have to `spy(oAuth2AuthorizedClientResolver)`

Makes sense?

Reply:
Ok I'll have a think about how it could be reworked.

Though I made it public as I need access in the authorizeWithClientCredentials method in ServerOAuthorizedCientExchangeFilterFunction not just for tests.

Reply:
on second review, instead of what I suggested previously for the test, let's just change the method signature to:

`Mono<OAuth2AuthorizedClient> clientCredentials` with package-private access

Reply:
Sounds good, i'll do that now